ta/pkcs11/src/processing_rsa.c

```diff
@@ -0,0 +1,544 @@
+			*tee_id = TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA224;
+			return PKCS11_CKR_OK;
+		}
+		if (hash == PKCS11_CKM_SHA256 &&
+		    mgf == PKCS11_CKG_MGF1_SHA256) {
+			*tee_id = TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA256;
+			return PKCS11_CKR_OK;
+		}
+		if (hash == PKCS11_CKM_SHA384 &&
+		    mgf == PKCS11_CKG_MGF1_SHA384) {
+			*tee_id = TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA384;
+			return PKCS11_CKR_OK;
+		}
+		if (hash == PKCS11_CKM_SHA512 &&
+		    mgf == PKCS11_CKG_MGF1_SHA512) {
+			*tee_id = TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA512;
+			return PKCS11_CKR_OK;
+		}
+		return PKCS11_CKR_MECHANISM_PARAM_INVALID;
+	}
+
+	switch (*tee_id) {
+	case TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA1:
+		if (hash != PKCS11_CKM_SHA_1 || mgf != PKCS11_CKG_MGF1_SHA1)
+			return PKCS11_CKR_MECHANISM_PARAM_INVALID;
+		break;
+	case TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA224:
+		if (hash != PKCS11_CKM_SHA224 || mgf != PKCS11_CKG_MGF1_SHA224)
+			return PKCS11_CKR_MECHANISM_PARAM_INVALID;
+		break;
+	case TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA256:
+		if (hash != PKCS11_CKM_SHA256 || mgf != PKCS11_CKG_MGF1_SHA256)
+			return PKCS11_CKR_MECHANISM_PARAM_INVALID;
+		break;
+	case TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA384:
+		if (hash != PKCS11_CKM_SHA384 || mgf != PKCS11_CKG_MGF1_SHA384)
+			return PKCS11_CKR_MECHANISM_PARAM_INVALID;
+		break;
+	case TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA512:
+		if (hash != PKCS11_CKM_SHA512 || mgf != PKCS11_CKG_MGF1_SHA512)
+			return PKCS11_CKR_MECHANISM_PARAM_INVALID;
+		break;
+	default:
+		return PKCS11_CKR_GENERAL_ERROR;
+	}
+
+	return PKCS11_CKR_OK;
+}
+
+enum pkcs11_rc
+pkcs2tee_proc_params_rsa_oaep(struct active_processing *proc,
+			      struct pkcs11_attribute_head *proc_params)
+{
+	struct serialargs args = { };
+	enum pkcs11_rc rc = PKCS11_CKR_GENERAL_ERROR;
+	struct rsa_oaep_processing_ctx *ctx = NULL;
+	uint32_t hash = 0;
+	uint32_t mgf = 0;
+	uint32_t source_type = 0;
+	void *source_data = NULL;
+	uint32_t source_size = 0;
+
+	serialargs_init(&args, proc_params->data, proc_params->size);
+
+	rc = serialargs_get(&args, &hash, sizeof(uint32_t));
+	if (rc)
+		return rc;
+
+	rc = serialargs_get(&args, &mgf, sizeof(uint32_t));
+	if (rc)
+		return rc;
+
+	rc = serialargs_get(&args, &source_type, sizeof(uint32_t));
+	if (rc)
+		return rc;
+
+	rc = serialargs_get(&args, &source_size, sizeof(uint32_t));
+	if (rc)
+		return rc;
+
+	rc = serialargs_get_ptr(&args, &source_data, source_size);
+	if (rc)
+		return rc;
+
+	if (serialargs_remaining_bytes(&args))
+		return PKCS11_CKR_ARGUMENTS_BAD;
+
+	proc->extra_ctx = TEE_Malloc(sizeof(struct rsa_oaep_processing_ctx) +
+				     source_size,
+				     TEE_USER_MEM_HINT_NO_FILL_ZERO);
+	if (!proc->extra_ctx)
+		return PKCS11_CKR_DEVICE_MEMORY;
+
+	ctx = proc->extra_ctx;
+
+	ctx->hash_alg = hash;
+	ctx->mgf_type = mgf;
+	ctx->source_type = source_type;
+	ctx->source_data_len = source_size;
+	TEE_MemMove(ctx->source_data, source_data, source_size);
+
+	return PKCS11_CKR_OK;
+}
+
+enum pkcs11_rc
```

Comment:
For consistency, could add an inline description:
```
/*
 * Set TEE RSA OAEP algorithm identifier upon PKCS11 mechanism parameters 
 * @tee_id: output TEE RSA OAEP algorithm identifier
 * @tee_hash_id: output TEE hash algorithm identifier
 * @proc_params: PKCS11 processing parameters
 */

Reply:
OK